### DISCUSSION TITLE
Open global Valhalla server landed!

### DISCUSSION THREAD

**Discussion by nilsnolde** [valhalla/valhalla#3373] [Show and tell]
[FOSSGIS e.V.](https://www.fossgis.de) approved our (@TimMcCauley, @kevinkreiser and me) [proposal](https://www.fossgis.de/wiki/Förderanträge/Valhalla-Routingserver) (on German only, sorry..) to fund a global Valhalla server, for at least one year. Among other open-source services, they already host the global OSRM demo service since Mapbox shut down theirs, which is also used on openstreetmap.org.

On the technical side we'll run one server continuously building tiles for the planet (at around a 2-day interval since there's "only" 16 threads on the server). As a "service" server answering requests we'll only deploy a single server for now and await users' feedback. That'll need fairly strict rate limiting, so it won't be usable for any production service for third parties. However, we do hope to eventually bring back the service to openstreetmap.org as a third alternative, where it has been before, in Mapzen days. The feasibility and acceptance of that proposal will be determined further down the road.

We aim to have all endpoints live except for `/transit_available` and verbose `/status`. Limits will have to be fairly strict for the computationally expensive ones like `/expansion`, `/centroid` or `/matrix`, the others will be kept at generous levels. The Tilezen/Nextzen elevation tiles from [Amazon Open Data](https://registry.opendata.aws/terrain-tiles/) (kudos where they are deserved) will be available for `/height` & costing.

That's great news, and we're very grateful to FOSSGIS e.V. for the funding. It'll be an important step to increase Valhalla's reach in the community, hopefully attracting more contributors along the way.

**Comment by nilsnolde** [comment]
UPDATE:

sometime last week we finally finished the server & software setup, Valhalla is now available at https://valhalla1.openstreetmap.de!

We published a small news article on German on [FOSSGIS news](https://fossgis.de/news/2021-11-12_funding_valhalla/) (English coming up soon).

The gist is: we rate limit to 1 call/user/sec and 100 calls/sec total. Service limits are a bit stricter than the defaults as well. That's about all one needs to know.

Happy routing!

**Reply by TimMcCauley** [reply]
Awesome news @nilsnolde !!!

**Reply by buchasia** [reply]
Hallo @nilsnolde, 

the `/height` endpoint is still not active. Will it be included in this server? 

Thank you!

**Reply by nilsnolde** [reply]
Yeah we could do that soonish.. never used it personally, but we got the disk space for it. @kevinkreiser you think it’s ok for our resources? Not sure what to expect in terms of performance hit without looking into it deeper.

**Reply by kevinkreiser** [reply]
@nilsnolde its incredibly performant if you use uncompressed sources, in that mode you will not even notice it impacting other requests

**Reply by nilsnolde** [reply]
Ok perfect, thought so. we can use uncompressed tiles, it has a few TB.

**Reply by nilsnolde** [reply]
Ok perfect, thought so. I’ll transfer the builders tiles then to the service I’m the next days. We can use uncompressed tiles, it has a few TB. Thanks @buchasia kinda forgot about it!

**Comment by buchasia** [comment]
Thank you @nilsnolde and @kevinkreiser :)

**Reply by nilsnolde** [reply]
quick update: started transferring elevation tiles, so should be available later today unless I forget again;) if it's not done by beginning of next week please do remind me here

**Reply by nilsnolde** [reply]
ok done. new instance is starting in 1-2 hours, latest tonight you can try. do note, we only provide elevation for areas where valhalla built tiles, i.e. ocean etc is not (yet) working.

**Reply by nilsnolde** [reply]
and as a thanks the server crashed due to open file limit :roll_eyes: might take a while to get it sorted, can't ssh into it anymore so FOSSGIS has to hard reboot

**Reply by kevinkreiser** [reply]
doh sorry i totally didnt think to warn you about setting the `ulimit`s

**Reply by buchasia** [reply]
@nilsnolde I hope it did not lead to a lot of work for you?

**Reply by nilsnolde** [reply]
it did, but it's not your fault;) should be online again today, we only could re-configure the server yesterday.

**Reply by nilsnolde** [reply]
it's up again and `/height` is active :tada: 

```
curl --request GET \
  --url https://valhalla1.openstreetmap.de/height \
  --header 'Content-Type: application/json' \
  --data '{
	"shape": [
		{
			"lon": 86.924932,
			"lat": 27.987892
		}
	]
}'
```

**Reply by zstadler** [reply]
The `/height` endpoint is not working now

```
curl --request GET \
  --url https://valhalla1.openstreetmap.de/height \
  --header 'Content-Type: application/json' \
  --data '{
        "shape": [
                {
                        "lon": 86.924932,
                        "lat": 27.987892
                }
        ]
}'
{"shape":[{"lat":27.987892,"lon":86.924932}],"height":[null]}
```

as well as the elevation graph

[<img width="1920" height="879" alt="image" src="https://github.com/user-attachments/assets/4fbb36be-2979-449e-80f4-ca553cc49d30" />](https://valhalla.openstreetmap.de/directions?profile=car&wps=34.88441579625473%2C30.580155895912085%2C34.795412065448886%2C30.57924693548938%2C34.67016434635883%2C30.517919348536424)

**Reply by nilsnolde** [reply]
Yep that's on purpose. We'll turn it on in the next few weeks sometime.

**Comment by farfromrefug** [comment]
@nilsnolde i am releasing an app for iOS and Android which uses https://valhalla1.openstreetmap.de for elevation profile and some route stats. Are you ok with that? Do you need me to add any header? 
I do use Cache Control to try and limit the calls.

**Reply by nilsnolde** [reply]
A proper User-Agent header is always appreciated, esp for these kind of "mass" deployments. I'd just be curious if it'll be OSS?

**Reply by farfromrefug** [reply]
@nilsnolde yes it is oss. https://github.com/Akylas/alpimaps. I Will add a User-Agent with `AlpiMaps`

**Reply by nilsnolde** [reply]
Cool, please go ahead:)

**Comment by mlidodo** [comment]
@nilsnolde We're trying to roll out our own Valhalla instance. Currently we're struggling with proper configuration as we occasionally get 422 (route not found) from Matrix API. The same matrix works on this global Valhalla instance (we're using most recent data from Geofabrik).

Could you share the configuration this Valhalla server uses? I haven't been able to find it online. The configuration does not seem to contain any sensitive information (maybe path to files?), but if it's not possible to share entire configuration, could you please share at least Loki configuration (service_defaults)?

Thanks a lot!

**Reply by chrstnbwnkl** [reply]
The config is not that different from the defaults, I suspect setting `thor.costmatrix_check_reverse_connection` and `thor.costmatrix_allow_second_pass` to true will solve your problems.

**Reply by mlidodo** [reply]
That works. What also works is increasing `service_defaults.radius` in Loki configuration. Which solution is better? Wouldn't setting the two mentioned values in Thor config increase duration of the matrix calculation? Increasing the radius for finding the nearest location on the route network should only affect the initial phase of the matrix calculation request, right?

**Reply by chrstnbwnkl** [reply]
It will and it does :smile: I'd say the reverse connection check should be enabled, we've had some really strange results without it but only found out after the default value was agreed upon.

**Comment by wafisher** [comment]
Hi @nilsnolde is there anyone who maintains an enterprise/paid version of the Valhalla API? We'd love to pay to avoid rate limits but don't want to roll our own.

I couldn't find any such service online, alas.

**Reply by drewda** [reply]
There are a number of firms that sell access to Valhalla without necessarily mentioning the software being used behind the scenes.

At Interline we do mention that it's Valhalla that we host. There's self-serve access for on-demand amounts of usage via https://rapidapi.com/interline-technologies-interline-technologies-default/api/interline-global-valhalla-navigation-and-routing-engine or discounted bulk subscriptions are available via https://app.interline.io/contact_forms/enterprise_inquiry?products[]=interline_routing_platform

**Reply by wafisher** [reply]
Works super well for anyone else interested! (Just signed up)

**Reply by nilsnolde** [reply]
@wafisher we should probably have a page to promote companies/services that offer Valhalla related stuff, there's enough that even contribute regularly. One of those is https://stadiamaps.com/products/routing-navigation/, great guys with reliable services.

**Comment by lll06270628-sketch** [comment]
Hi Valhalla/FOSSGIS team — TripReadyGo (https://tripreadygo.com) is starting a small, friends-only internal test. For Korean street-exploration requests, Google Places selects and verifies POIs and Google Routes is tried first; only when Google returns no pedestrian route does our backend call the public Valhalla demo for pedestrian road geometry, distance, and duration. Requests send `X-Client-Id: tripreadygo.com`, use a client-side minimum interval of 1.1 seconds, and display OpenStreetMap/Valhalla attribution. This is low-volume internal testing, not a long-term production SLA dependency. We will choose a controlled hosted or self-hosted route service before scaling. Please let us know if you prefer different identification or limits.

**Comment by tchobanski** [comment]
Title: Buchungsverwaltung (BuVer) — small business app, X-Client-Id: buchungsverwaltung

Hi, and thank you for running this instance.

Following the request in this discussion to announce published applications and to identify our requests, here is what we are and what we send.

What the application is

Buchungsverwaltung ("BuVer") — a small multi-tenant business application used by a handful of organisations in Austria. It covers a cash book, a vehicle logbook, time tracking and facility management. Not a public service, not a consumer app: every request behind it is a logged-in employee who typed something.

Where Valhalla is used

Exactly one place: the vehicle logbook. When a driver ends a trip, the application computes the driving distance between the recorded start and end position and suggests an odometer reading. The driver can override it.

It also uses /route in an on-demand "route calculator" tab, on button press.

What we send

Server-side only. The browser never talks to your instance — it calls our own API, and our backend calls yours. You see one server IP, not our users'.
Two requests per calculation: the default auto costing with alternates: 2, and a second one with costing_options.auto.shortest for the shortest route. We need both because the shortest route is genuinely different from the alternates — measured over ten Austrian routes, the shortest route was never among the alternates, differing by up to 25.5 km.
Sequential, not parallel. Until 2026-09-04 we issued those two requests concurrently. We changed that after reading the 1 request/user/second figure here. Measured cost of serialising: +102 ms on the longest route in the country — nobody notices.
No caching. Each calculation goes out.
Headers:

X-Client-Id: buchungsverwaltung
User-Agent: Buchungsverwaltung/1.0
Volume

Small, and I would rather give you the real number than a comfortable one. Across all tenants of our instance:

Month	Trips ended (= route calculations)
2026-08	14
2026-09 (partial)	7
That is roughly 30–60 requests per month in total. Seven users currently have the logbook module. Concurrency is effectively one calculation at a time.

Where we are heading

We are aware of your note that the demo server "won't be usable for any production service for third parties". We run our own Valhalla instance already, and tenants can select it; the public instance is currently the default and the fallback. Moving self-hosted to the default is on our list.

Two more things, in case they are useful to you:

On 2026-09-02 the instance was unreachable for us for a full day (no open port on 80 or 443 from our host, DNS resolving to 162.55.103.19), while Nominatim next door answered in 27 ms. That is what prompted us to build a fallback to OSRM rather than leave users without a distance. Not a complaint — just a data point about what an outage looks like from outside.
The rate limit does not appear to be enforced today: ten concurrent /route requests all returned 200, with no X-RateLimit-* or Retry-After headers. We are keeping to one at a time anyway.
Happy to reduce, cache, or move off entirely if our pattern is a problem for you — just say so here and we will.

**Reply by nilsnolde** [reply]
Thanks!

> The rate limit does not appear to be enforced today: ten concurrent /route requests all returned 200

If you had tried the 11th time it'd have blocked:) we do allow for _bursts_ of 10.

And yeah it was down. Service limits are a lot stricter now, _shouldnt_ happen too much anymore. It's the cost of running the full planet on a low spec machine.

**Comment by elazaroo** [comment]
Title: Fuel Calculator — Android app (Spain), X-Client-Id: fuel-calculator-android

Hi, and thank you for running this instance.

Following the request in this discussion to announce published applications and to identify their requests, here is ours.

**What the application is**

Fuel Calculator, a free Android app (no ads, no accounts) about to be published on Google Play, for Spain only. It ranks nearby fuel stations by the real total cost of refuelling, i.e. fuel price plus the fuel spent driving there and back, and estimates the cost of a car trip.

**Where Valhalla is used**

Always on a user action (opening the stations tab or pressing "Calcular"), never in the background:

- `/sources_to_targets`: one request per station search. 1 source (the user's position), at most 25 targets within a 1–50 km radius (10 km by default), `auto` costing. It gives real road distances instead of straight lines.
- `/route`: only when the user calculates a trip with "avoid tolls" (the default). 2–4 locations, `costing_options.auto.use_tolls: 0`, `directions_type: none`. Trips with tolls go to a commercial provider, not to you.

**What we send**

- Requests go straight from each user's phone, so every user counts separately against the per-user limit.
- Header on every request: `X-Client-Id: fuel-calculator-android`.
- One request at a time, no retries in a loop. Prices and results are cached on the device, but routes and matrices are not.

**Volume**

The app isn't live yet, so there is no real number to share. A typical session is 1 matrix request plus 0–2 route requests. We'll post an update here once there is real usage data.

**Fallbacks and where we are heading**

We read your note that the demo server "won't be usable for any production service for third parties". If the instance is down or rate-limits us, the app keeps working: station distances fall back to straight-line estimates and toll-free trips are routed by the commercial provider. If usage grows, we'll move the matrix calls to our own instance.

If this pattern is a problem for you, please say so here and we'll reduce it, cache it or move off. Contact: @elazaroo.

**Comment by jsuchyvetter** [comment]
RouteStudio – internal bus route planning tool (Vetter Verkehrsbetriebe, Germany)

Hi,

thanks a lot for running the public Valhalla server!

We'd like to let you know that we have started using it from RouteStudio, an internal web tool of Vetter Verkehrsbetriebe, a regional bus operator in Saxony-Anhalt, Germany. We use it to plan bus diversions during road works and to create passenger information for bus stops.

How we use the server:

route only, with costing: "bus" and vehicle dimensions (articulated bus), and break_through waypoints so that routes do not U-turn at intermediate points.
Requests are only sent when a planner edits a line manually (draws it, moves or inserts a point). Computed routes are stored with the project and are not requested again when it is reopened.
Client-side throttling: at most 1 request per second, sent one after another.
Every request includes the header X-Client-Id: routestudio-vetter.
Around 4 internal users, about 200 requests on a very busy day. Traffic is limited to our operating area in Saxony-Anhalt.
If the server is not available, the tool falls back to a different routing service, so a downtime does not block our work.
The tool is not public and is not offered as a service to third parties. If this use doesn't fit your fair-use policy, or if you'd like us to change something (request volume, headers, etc.), please let us know here. We'd be happy to adjust.

Thanks again for providing this service to the community!

Jan
Vetter Verkehrsbetriebe

**Comment by Desparator13** [comment]
Hi FOSSGIS team, thank you for running the public Valhalla server.

I'm a volunteer in Bangkok. Since 26 September 2026 Bangkok has been under a flood
disaster declaration, and I've built a free, non-commercial web map to help people
find a way home that avoids flooded roads:
https://bkk-flood-map.bkk-flood-map.workers.dev

How it uses valhalla1.openstreetmap.de:
- Requests go directly from each user's browser (not proxied through a server),
  with the header `X-Client-Id: bkk-flood-map`.
- The client waits at least 1.15 s between requests (the 1 req/s limit).
- To avoid flooded roads we call /route with `exclude_polygons` built from current
  flood reports, keeping the total perimeter under 9.5 km. One route search uses
  at most 5 requests, or up to 10 when it also checks an expressway alternative.
- In-app turn-by-turn navigation only re-requests a route when the driver leaves
  the route (60 m+, at most once every 20 s) or a new flood report blocks the road ahead.
- No commercial use and no ads. Valhalla, FOSSGIS and OpenStreetMap are credited in the app.

This is for the flood emergency period (about 1–2 weeks). I can't predict the traffic
precisely. If the load becomes a problem, or you'd like anything changed (lower rate,
different header, or moving to our own instance), please reply here or email
ratchathammanoon.prompornchai@gmail.com and I'll adjust right away.

Thank you!
